runelite-client/src/main/java/net/runelite/client/plugins/kingdomofmiscellania/KingdomConfig.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2020, Brandt Hill <https://github.com/BrandtHill>
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * 1. Redistributions of source code must retain the above copyright notice, this
+ *    list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright notice,
+ *    this list of conditions and the following disclaimer in the documentation
+ *    and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.kingdomofmiscellania;
+
+import java.time.Instant;
+import net.runelite.client.config.Config;
+import net.runelite.client.config.ConfigGroup;
+import net.runelite.client.config.ConfigItem;
+import net.runelite.client.config.Range;
+
+@ConfigGroup("kingdomOfMiscellania")
+public interface KingdomConfig extends Config
+{
+	int MAX_COFFER = 7_500_000;
+	int MAX_FAVOR_PERCENT = 100;
+
+	@ConfigItem(
+		position = 1,
+		keyName = "sendNotifications",
+		name = "Send Notifications",
+		description = "Send chat notifications upon login showing current estimated coffer and favor"
+	)
+	default boolean shouldSendNotifications()
+	{
+		return true;
+	}
+
+	@Range(
+		max = MAX_COFFER
+	)
+	@ConfigItem(
+		position = 2,
+		keyName = "cofferThreshold",
+		name = "Coffer Threshold",
+		description = "Send notifications if coffer is at or below this value"
+	)
+	default int getCofferThreshold()
+	{
+		return MAX_COFFER;
+	}
+
+	@Range(
+		max = MAX_FAVOR_PERCENT
+	)
+	@ConfigItem(
+		position = 3,
+		keyName = "favorThreshold",
+		name = "Favor Threshold",
+		description = "Send notifications if favor percentage is at or below this value"
+	)
+	default int getFavorThreshold()
+	{
+		return MAX_FAVOR_PERCENT;
+	}
+
+	/**
+	 * Stored data
+	 **/
+	@ConfigItem(

```

Comment:
This and everything below here is per-profile, and so should use the set/get rsprofile functions in config manager, so that it works correctly when using more than one account/profile.